aptos-move/move-examples/token_objects/sources/soul_bound_token.move

```diff
@@ -0,0 +1,110 @@
+module token_objects::soul_bound_token {
+    use std::string::String;
+    use token_objects::token::{MutabilityConfig, Token};
+    use std::signer;
+    use aptos_framework::object::{generate_transfer_ref, Object, disable_ungated_transfer, generate_linear_transfer_ref, object_from_constructor_ref, transfer};
+    use token_objects::token;
+    use aptos_framework::object;
+    use std::string;
+    use token_objects::collection;
+    use std::option;
+
+    struct OnChainConfig has key {
+        collection: String,
+        mutability_config: MutabilityConfig,
+    }
+
+    fun init_module(account: &signer) {
+        let collection = string::utf8(b"Mixed Tokens");
+        collection::create_aggregable_collection(
+            account,
+            string::utf8(b"collection description"),
+            collection::create_mutability_config(false, false),
+            *&collection,
+            option::none(),
+            string::utf8(b"collection uri"),
+        );
+
+        let on_chain_config = OnChainConfig {
+            collection: string::utf8(b"Souler"),
+            mutability_config: token::create_mutability_config(true, true, true),
+        };
+        move_to(account, on_chain_config);
+    }
+
+    public fun mint_soul_bound_token(
+        creator: &signer,
+        description: String,
+        name: String,
+        uri: String,
+        to: address,
+    ): Object<Token> acquires OnChainConfig {
```

Comment:
Do we want to make this a formal spec on how to do soul bound? Do we need this in the framework?

Let's say I make a new token and I want to add the features of it being soul bound, is there a way I can add an explicit SoulBound struct? Should it be SoulBound to a specific individual so that it can be generated and stored anywhere upon creation but can only be transferred to them? How do we think about nested SoulBound?

There's nothing fundamentally crazy about this example, but it would be great to add some documentation to the top.